src/objects/target.gd

```diff
@@ -21,8 +20,6 @@ func handle_hit(player_id: int) -> void:
 	if get_multiplayer().has_multiplayer_peer():
 		if get_multiplayer().is_server():
 			rpc("destroy_self", player_id)
-		else:
-			destroy_self(player_id)
```

Comment:
Only the server is responsible for destroying the targets. On the clients, their targets will be destroyed by the rpc call